core/src/main/java/jenkins/security/ApiTokenProperty.java

```diff
@@ -138,36 +195,120 @@ private boolean hasPermissionToSeeToken() {
-        //return User.idStrategy().equals(user.getId(), current.getId());
-        return StringUtils.equals(user.getId(), current.getId());
+        
+        return User.idStrategy().equals(user.getId(), current.getId());
     }
+    
+    // only for Jelly
+    @Restricted(NoExternalUse.class)
+    public List<ApiTokenStore.HashedToken> getTokenList() {
+        return tokenStore.getTokenListSortedByName();
+    }
+    
+    // only for Jelly
+    @Restricted(NoExternalUse.class)
+    public boolean mustDisplayLegacyApiToken() {
+        return apiToken != null || !ApiTokenPropertyConfiguration.get().isCreationOfLegacyTokenDisabled();
+    }
+    
+    @Override
+    public UserProperty reconfigure(StaplerRequest req, @CheckForNull JSONObject form) throws FormException {
+        if(form == null){
+            return this;
+        }
 
+        Object tokenStoreData = form.get("tokenStore");
+        Map<String, JSONObject> tokenStoreTypedData = convertToTokenMap(tokenStoreData);
+        this.tokenStore.reconfigure(tokenStoreTypedData);
+        return this;
+    }
+    
+    private Map<String, JSONObject> convertToTokenMap(Object tokenStoreData) {
+        if (tokenStoreData == null) {
+            // in case there are no token
+            return Collections.emptyMap();
+        } else if (tokenStoreData instanceof JSONObject) {
+            // in case there is only one token
+            JSONObject singleTokenData = (JSONObject) tokenStoreData;
+            Map<String, JSONObject> result = new HashMap<>();
+            addJSONTokenIntoMap(result, singleTokenData);
+            return result;
+        } else if (tokenStoreData instanceof JSONArray) {
+            // in case there are multiple tokens
+            JSONArray tokenArray = ((JSONArray) tokenStoreData);
+            Map<String, JSONObject> result = new HashMap<>();
+            for (int i = 0; i < tokenArray.size(); i++) {
+                JSONObject tokenData = tokenArray.getJSONObject(i);
+                addJSONTokenIntoMap(result, tokenData);
+            }
+            return result;
+        }
+        
+        throw HttpResponses.error(400, "Unexpected class received for the token store information");
+    }
+    
+    private void addJSONTokenIntoMap(Map<String, JSONObject> tokenMap, JSONObject tokenData) {
+        String uuid = tokenData.getString("tokenUuid");
+        tokenMap.put(uuid, tokenData);
+    }
+    
+    /**
+     * Only usable if the user still has his API token. After the revocation of the legacy token the method will do nothing.
+     * @deprecated Each token can be revoked now and new tokens can be requested without altering existing ones.
+     */
+    @Deprecated
     public void changeApiToken() throws IOException {
+        // just to keep the same level of security
         user.checkPermission(Jenkins.ADMINISTER);
+
+        LOGGER.log(Level.WARNING, "Deprecated usage of changeApiToken");
```

Comment:
I think that could help developer of plugin that are using directly the method to have a warning instead of silently doing / not doing things.

If you think it's useless, I will remove it.